src/vmm/src/persist.rs

```diff
@@ -131,6 +138,12 @@ impl Display for CreateSnapshotError {
             MicrovmState(err) => write!(f, "Cannot save microvm state: {}", err),
             SerializeMicrovmState(err) => write!(f, "Cannot serialize MicrovmState: {:?}", err),
             SnapshotBackingFile(err) => write!(f, "Cannot open snapshot file: {:?}", err),
+            TooManyDevices(val) => write!(
+                f,
+                "Too many devices attached. The maximum number allowed \
+                 for the snapshot data version requested is {}.",
+                val
+            ),
```

Comment:
Would this be more helpful?

```suggestion
            TooManyDevices(val) => write!(
                f,
                "Too many devices attached: {}. The maximum number allowed \
                 for the snapshot data version requested is {}.",
                val, FC_V0_23_MAX_DEVICES
            ),
```

Reply:
Done.